cmd/guardian/main.go

```diff
@@ -135,6 +141,33 @@ func main() {
 
 func realMain(ctx context.Context) error {
 	setLogEnvVars()
+
+	start := time.Now()
+	mClient, err := metrics.New(ctx, version.Name, version.Version)
+	if err != nil {
+		fmt.Printf("metric client creation failed: %v\n", err)
+	}
+
+	ctx = metrics.WithClient(ctx, mClient)
+	defer func() {
+		if r := recover(); r != nil {
+			handler := metricswrap.WriteMetric(ctx, mClient, "panics", 1)
+			defer handler()
+			panic(r)
+		}
+	}()
+
+	cleanup := metricswrap.WriteMetric(ctx, mClient, "runs", 1)
+	defer cleanup()
+
+	// This will cause a synchronous metrics call.
+	defer func() {
```

Comment:
This defer will run before the cleanup defer; is that intended?

Reply:
I guess that is a good point. I was thinking the order didn't really matter but we probably want the runtime to be all inclusive.